src/Collections-Atomic/WaitfreeQueue.class.st

```diff
@@ -295,3 +365,13 @@ WaitfreeQueue >> size [
 
 	^ count
 ]
+
+{ #category : #initialization }
+WaitfreeQueue >> withAll: aCollection [
```

Comment:
little note:
It would be better to name instance side method as #nextPutAll: to be closer to #nextPut: .

Reply:
> It would be better to name instance side method as #nextPutAll: to be closer to #nextPut: .

But the semantics are different:

- `#nextPut:` takes a value (entry) which is added to the queue in a thread-safe manner.
- `#withAll:` takes a collection of items (AtomicQueueItem) and adds them to the queue in a non-thread-safe manner.

It would of course be possible to write `#nextPutAll:`, but there hasn't been a need for it yet.